evalml/pipelines/components/transformers/preprocessing/text_featurizer.py

```diff
@@ -141,17 +140,17 @@ def transform(self, X, y=None):
             {s: "NaturalLanguage" for s in self._text_columns},
         )
         X_lsa = self._lsa.transform(X_ww_altered)
+        X_nlp_primitives.set_index(X_ww.index, inplace=True)
```

Comment:
Reset the index to be equal to the input X

Reply:
Just curious, why is this necessary for this PR?

Reply:
During AutoMLSearch, when we do the data splits, we get input indices that aren't in consecutive order. Seems like `X_nlp_primitives` completely resets the index (so it becomes 0, 1, 2, ...), which causes errors when we try to set the `None` values using the mask. I just moved this line from below to above the `for` loop so that the `.loc` method doesn't fail

Reply:
We don't have to do it for the lsa features because they are computed by the lsa component, which preserves the index because our components preserve indices!!